internal/nginx/config/maps_test.go

```diff
@@ -0,0 +1,184 @@
+							Add: []dataplane.HTTPHeader{
+								{
+									Name:  "my-add-header",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+				{
+					Filters: dataplane.Filters{
+						RequestHeaderModifiers: &dataplane.HTTPHeaderFilter{
+							Add: []dataplane.HTTPHeader{
+								{
+									Name:  "my-second-add-header",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+				{
+					Filters: dataplane.Filters{
+						RequestHeaderModifiers: &dataplane.HTTPHeaderFilter{
+							Set: []dataplane.HTTPHeader{
+								{
+									Name:  "my-set-header",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	conf := dataplane.Configuration{
+		HTTPServers: []dataplane.VirtualServer{
+			{
+				PathRules: pathRules,
+			},
+			{
+				PathRules: pathRules,
+			},
+			{
+				IsDefault: true,
+			},
+		},
+		SSLServers: []dataplane.VirtualServer{
+			{
+				PathRules: pathRules,
+			},
+			{
+				IsDefault: true,
+			},
+		},
+	}
+
+	expSubStrings := map[string]int{
+		"map ${http_my_second_add_header} $my_second_add_header_header_var {": 1,
+		"~.* ${http_my_second_add_header},;":                                  1,
+		"map ${http_my_set_header} $my_set_header_header_var {":               0,
+		"map ${http_my_add_header} $my_add_header_header_var {":               1,
+		"default '';":                 2,
+		"~.* ${http_my_add_header},;": 1,
+	}
+
+	maps := string(executeMaps(conf))
+	for expSubStr, expCount := range expSubStrings {
+		if expCount != strings.Count(maps, expSubStr) {
+			t.Errorf(
+				"executeMaps() did not generate maps with substring %q %d times. Maps: %v",
+				expSubStr,
+				expCount,
+				maps,
+			)
+		}
+	}
+}
+
+func TestBuildAddHeaderMaps(t *testing.T) {
+	g := NewGomegaWithT(t)
+	pathRules := []dataplane.PathRule{
+		{
+			MatchRules: []dataplane.MatchRule{
+				{
+					Filters: dataplane.Filters{
+						RequestHeaderModifiers: &dataplane.HTTPHeaderFilter{
+							Add: []dataplane.HTTPHeader{
+								{
+									Name:  "my-add-header",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+				{
+					Filters: dataplane.Filters{
+						RequestHeaderModifiers: &dataplane.HTTPHeaderFilter{
+							Add: []dataplane.HTTPHeader{
+								{
+									Name:  "my-second-add-header-extra-long-name-here",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+				{
+					Filters: dataplane.Filters{
+						RequestHeaderModifiers: &dataplane.HTTPHeaderFilter{
+							Set: []dataplane.HTTPHeader{
+								{
+									Name:  "my-set-header",
+									Value: "some-value-123",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testServers := []dataplane.VirtualServer{
+		{
+			PathRules: pathRules,
+		},
+		{
+			PathRules: pathRules,
+		},
+		{
+			IsDefault: true,
+		},
+	}
+	resultMap := []http.Map{
```

Comment:
nit: better to rename to `expectedMap` for consistency with the other tests. Also, result can also mean the result of the function call under the test.